### ISSUE TITLE
C# incorrect interpretation of '?'-operators

### ISSUE THREAD

**Issue by rxDietel** [doxygen/doxygen#8957]
### Describe the bug
Properties with ternary operators or null-conditional operators will have a returntype with as much '?' added as there are '?' in the method.

### Expected behavior
Properties types don't have '?' appended unless they are nullable

### Screenshots
![grafik](https://user-images.githubusercontent.com/94047927/145405960-5ab58756-6018-4100-b5a8-653287d52f96.png)

### To Reproduce
`
        /// <summary>
        ///     Gets the endpoint address.
        /// </summary>
        public EndpointAddress EndpointAddress
        {
            get
            {
                var created = Uri.TryCreate(NetworkSettings?.ServerIp, UriKind.Absolute, out var systemUri);
                return new EndpointAddress(created ? systemUri : new Uri(DefaultServerUri));
            }
        }
`

### Version
![grafik](https://user-images.githubusercontent.com/94047927/145406217-b49e26b7-09ae-45ac-a0da-59c06b78ed02.png)
Windows 10 64bit
used setup from website

**Comment by albert-github** [comment]
"used setup from website" is a bit vague but with a default setup (`doxygen -g`) I do see the 2 question marks as well.
Normally always (the standard):
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by rxDietel** [comment]
Thank you for reacting so fast!
Should I still create a example for reproducing the problem?

**Comment by albert-github** [comment]
I think it still would be nice, though even with the limited information available I think I was able to reproduce the problem already.
The extra example would help to find maybe even more problems with question marks.

**Comment by albert-github** [comment]
In the code I see `NetworkSettings?.ServerIp` is this correct, can an identifier in C# contain a question mark or what is otherwise here the meaning of the question mark?
The question mark in the return statement is a normal operator and OK to me (though causes also some problems in doxygen).

**Comment by rxDietel** [comment]
thats a null-conditional-operator.
it will only atempt the action behind the '.' if the identifier in front of '?' != null
[microsoft documentation for operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-)

```
if(thing != null)
  return thing.property;
else
  return null;
```
can be written as:
```
return thing?.property;
```

**Comment by rxDietel** [comment]
I hope i will find some time tomorrow to create some example code.
probably using some more operators with '?' in them.

**Comment by albert-github** [comment]
I understand the usage of the `?` and I have an idea how to solve it (but there might be other cases as well where it goes wrong or will go wrong with the fix I have in mind. I hope to create a proposed patch later today). So definitely some more cases would really be useful.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8959

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by rxDietel** [comment]
Tested the new release and seems to work as expected.
